src/Controller/ProductAnalysisController.php

```diff
@@ -2,40 +2,61 @@
 
 namespace App\Controller;
 
+use App\Repository\BestSellingProductRepository;
 use App\Service\AIService;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
-use Symfony\Component\HttpFoundation\JsonResponse;
-use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class ProductAnalysisController extends AbstractController
 {
     private $aiService;
```

Comment:
imo you should also declare $repository explicitly: ```private $repository;```

Reply:
Also, if your prestashop supports PHP 8, you could use[ Constructor property promotion](https://stitcher.io/blog/constructor-promotion-in-php-8): 
```suggestion
 public function __construct(
        private AIService $aiService,
        private BestSellingProductRepository $repository
    ) {}
```


Test for yourself 😄